src/main/java/com/keniu/models/ShoppingCart.java

```diff
@@ -0,0 +1,40 @@
+package com.keniu.models;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted = false")
+@Getter
+@Setter
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @JoinColumn(nullable = false)
+    @OneToOne
+    private User user;
```

Comment:
I believe that the shopping cart cannot exist without the user, so we can combine the primary key and foreign key in one column to avoid an extra column and index for it.

P.S. Don't forget to change your scripts
```suggestion
    @Id
    private Long id;
    
    @MapsId
    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
```

Reply:
Ok, done